test/suite/lib.rs

```diff
@@ -218,13 +219,80 @@ pub async fn test_supported_standards(ledger: impl LedgerEnv) -> anyhow::Result<
     Ok(Outcome::Passed)
 }
 
+/// Checks whether the ledger advertizes support for ICRC-1 standard.
+pub async fn test_tx_deduplication(ledger_env: impl LedgerEnv) -> anyhow::Result<Outcome> {
+    // Create two test accounts and transfer some tokens to it
+    let p1_env = setup_test_account(&ledger_env, 100_000.into()).await?;
+    let p2_env = p1_env.fork();
+
+    // If created at time is not set, the transfer should not be calssified as a duplicate --> Transfer should succeed
+    let transfer_args = Transfer::amount_to(10_000, p2_env.principal());
+    assert!(
+        transfer(&p1_env, transfer_args.clone())
+            .await
+            .unwrap()
+            .unwrap()
+            < transfer(&p1_env, transfer_args.clone())
+                .await
+                .unwrap()
+                .unwrap()
+    );
```

Comment:
We must not panic in the test suite because this will crash the entire test process. We should return an `Outcome` instead.